test/unit/mdc-menu/simple.foundation.test.js

```diff
@@ -62,216 +94,358 @@ test('exports numbers', () => {
+testFoundation('#open anchors the menu to the bottom left in LTR when close to the bottom edge with margin',
   ({foundation, mockAdapter, mockRaf}) => {
     td.when(mockAdapter.hasAnchor()).thenReturn(true);
     td.when(mockAdapter.isRtl()).thenReturn(false);
     td.when(mockAdapter.getInnerDimensions()).thenReturn({height: 200, width: 100});
     td.when(mockAdapter.getWindowDimensions()).thenReturn({height: 1000, width: 1000});
     td.when(mockAdapter.getAnchorDimensions()).thenReturn({
-      height: 20, width: 40, top: 900, bottom: 20, left: 10, right: 50,
+      height: 20, width: 40, top: 900, bottom: 920, left: 10, right: 50,
     });
-    td.when(mockAdapter.getAccurateTime()).thenReturn(0);
-
+    foundation.setAnchorCorner(Corner.BOTTOM_START);
+    foundation.setAnchorMargin({top: 5, left: 7, bottom: 10, right: 0});
     foundation.open();
     mockRaf.flush();
-    mockRaf.flush();
+    td.verify(mockAdapter.setTransformOrigin('left bottom'));
+    td.verify(mockAdapter.setPosition({left: '7px', bottom: '15px'}));
+  });
 
-    td.when(mockAdapter.getAccurateTime()).thenReturn(500);
+testFoundation('#open anchors the menu to the bottom left in RTL when close to the bottom and right edge with margin',
+  ({foundation, mockAdapter, mockRaf}) => {
+    td.when(mockAdapter.hasAnchor()).thenReturn(true);
+    td.when(mockAdapter.isRtl()).thenReturn(true);
+    td.when(mockAdapter.getInnerDimensions()).thenReturn({height: 200, width: 100});
+    td.when(mockAdapter.getWindowDimensions()).thenReturn({height: 1000, width: 1000});
+    td.when(mockAdapter.getAnchorDimensions()).thenReturn({
+      height: 20, width: 40, top: 900, bottom: 920, left: 910, right: 950,
+    });
+    foundation.setAnchorCorner(Corner.BOTTOM_START);
+    foundation.setAnchorMargin({top: 5, left: 0, bottom: 10, right: 7});
+    foundation.open();
     mockRaf.flush();
-    td.verify(mockAdapter.setTransformOrigin('top left'));
-    td.verify(mockAdapter.setPosition({left: '0', top: '0'}));
+    td.verify(mockAdapter.setTransformOrigin('right bottom'));
+    td.verify(mockAdapter.setPosition({right: '7px', bottom: '15px'}));
   });
 
+testFoundation('#open the menu after setting invalid margins.',
+  ({foundation, mockAdapter, mockRaf}) => {
+    td.when(mockAdapter.hasAnchor()).thenReturn(true);
+    td.when(mockAdapter.isRtl()).thenReturn(true);
+    td.when(mockAdapter.getInnerDimensions()).thenReturn({height: 200, width: 100});
+    td.when(mockAdapter.getWindowDimensions()).thenReturn({height: 1000, width: 1000});
+    td.when(mockAdapter.getAnchorDimensions()).thenReturn({
+      height: 20, width: 40, top: 900, bottom: 920, left: 910, right: 950,
+    });
+    foundation.setAnchorCorner(Corner.BOTTOM_START);
+    foundation.setAnchorMargin({top: 5, left: 0, bottom: 10, right: 7});
+    foundation.setAnchorMargin({top: '5px', left: 'zz', bottom: '10dp', right: '7em'});
```

Comment:
Why two `setAnchorMargin` calls?

And what's the motivation for testing invalid input for this specific API? (I doubt we really do that in general...)

Reply:
code coverage is one of the motivations.

Reply:
Oh, for the `typeof` operations? This may become unnecessary then, if we revise the API RE my other comment (i.e. the branch this hits will no longer exist, and those branches are really intended for nonexistent values rather than invalid ones, right?).